### PR TITLE
Add install ccache step for MacOS

### LINKED ISSUE #665: Document how to install ccache for macOS and virtualenv
It would help contributors who are not too familiar with ccache to avoid long build times and also avoid issues with conda.

### PR DESCRIPTION
**Pull request by rauletorresc**
**Context:** Installing `ccache `via `brew `can avoid long build times.

**Description of the Change:** Add `ccache `installation via `brew`. Add hint on how to install `brew`.

**Benefits:** Shorter build times on MacOS.

Partially solves #382 

Closes #665 

[sc-67079]

### CODE REVIEW

**Comment by rauletorresc** [comment]
@erick-xanadu FYI I tested under MacOS 12 and the `ccache` drop-in compiler replacements worked well with AppleClang 14

**Comment by erick-xanadu** [comment]
Do we need rust?

**Comment by rauletorresc** [comment]
> Do we need rust?

According to https://github.com/PennyLaneAI/catalyst/issues/382, yes @josh146 what do you think? Is this still valid?:
"Make sure it is clear that for MacOS users, cmake, **rust**, and ninja have to be installed, as well as how to install XCode."

**Comment by erick-xanadu** [comment]
https://github.com/PennyLaneAI/catalyst/pull/470

**Comment by dime10** [comment]
It's true that we need homebrew to be installed, but maybe that should be a separate preliminary step "if you don't have homebrew yet, ..." since many users will already have it?

File: doc/dev/installation.rst [hunk 1]
```diff
@@ -111,8 +111,14 @@ The essential steps are:
 
         # Install XCode Command Line Tools and common requirements
         xcode-select --install
-        pip install cmake ninja
-        brew install libomp
+        pip install cmake ninja rust
+
+        # Install Homebrew before using it for installing other packages
+        /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
```
[the remark this answers is not in the document]
Comment (rauletorresc) [reply doc/dev/installation.rst]: Done

**Review by erick-xanadu — APPROVED** [review]

**Pull request merged**